test/lcov_test.dart

```diff
@@ -122,6 +124,37 @@ void main() {
       expect(res, contains(p.absolute(p.join('lib', 'src', 'util.dart'))));
     });
 
+    test('formatLcov() excludes files matching glob patterns', () async {
+      final hitmap = await _getHitMap();
+
+      final resolver = await Resolver.create(packagePath: '.');
+      final res = hitmap.formatLcov(
+        resolver,
+        ignoreGlobs: {Glob('**/*.g.dart'), Glob('**/util.dart')},
+      );
+
+      expect(res, isNot(contains(p.absolute(_sampleGeneratedPath))));
```

Comment:
I think you should be able to add an extra check for this file to all the other tests in this file: `expect(res, contains(p.absolute(_sampleGeneratedPath)));`. That way you're checking both the contains and doesn't contain cases.

At the moment I think these tests could pass accidentally by, for example, mistyping the `_sampleGeneratedPath`, since every test that checks it is checking that it's not in the report.

Reply:
I added this check as well as some additional checks in globs related tests to make the expected behaviour more explicit